ucp/_libs/utils.py

```diff
@@ -60,3 +66,64 @@ def print_multi(values, key_length=25):
     print_str = "".join(f"{s: <{key_length}} | " for s in values[:-1])
     print_str += values[-1]
     print(print_str)
+
+
+def get_address(ifname=None):
+    """
+    Get the address associated with a network interface.
+
+    Parameters
+    ----------
+    ifname : str
+        The network interface name to find the address for.
+        If None, it uses the value of environment variable `UCXPY_IFNAME`
+        and if `UCXPY_IFNAME` is not set it defaults to "ib0"
+        An OSError is raised for invalid interfaces.
+
+    Returns
+    -------
+    address : str
+        The inet addr associated with an interface.
+
+    Examples
+    --------
+    >>> get_address()
+    '10.33.225.160'
+
+    >>> get_address(ifname='lo')
+    '127.0.0.1'
+    """
+
+    def _get_address(ifname):
+        ifname = ifname.encode()
+        with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
+            return socket.inet_ntoa(
+                fcntl.ioctl(
+                    s.fileno(), 0x8915, struct.pack("256s", ifname[:15])  # SIOCGIFADDR
+                )[20:24]
+            )
+
+    def _try_interfaces():
```

Comment:
Do these need to be closures? Or could they be top-level functions that just get called here?

Reply:
They could, but they're exclusively meant to be part of the `get_address()` implementation, we don't want people to be calling them directly by accident. Are there any specific reasons to suggest this change?

Note: I'm just moving the code around, not changing the implementation in this PR.

Reply:
No more thoughts from me